### ISSUE TITLE
In with way is merged gradient in graph multiple outputs

### ISSUE THREAD

**Issue by jllombart** [fchollet/keras#1924]
I'm trying perform some one input to multiple output models
This is accomplished with graph model like here
```
model = Graph()
model.add_input(name='input',input_shape=(784,))
model.add_node(Dense(512,activation='relu'),name='c1',input='input')
model.add_node(Dense(512,activation='relu'),name='c2',input='c1')
model.add_node(Dense(512,activation='relu'),name='c3',input='c2')
model.add_node(Dense(512,activation='relu'),name='c4',input='c3')


model.add_node(Dense(512,activation='relu'),name='l5',input='l4')
model.add_node(Dense(512,activation='relu'),name='l6',input='l5')
model.add_node(Dense(10,activation='softmax'),name='sl1',input='l6')
model.add_output(name='out',input='sl1')

model.add_node(Dense(256,activation='relu'),name='r5',input='l4')
model.add_node(Dense(256,activation='relu'),name='r6',input='r5')
model.add_node(Dense(2,activation='softmax'),name='sr1',input='r6')
model.add_output(name='out2',input='sr1')


rms = RMSprop()
model.compile(rms,{'out':'categorical_crossentropy','out2':'categorical_crossentropy'})
```

This works fine for me but the gradient is merged in the common node ( in examale in l4 ), like the sum of the gradients of each side of the graph

My question is how or where I can change this way to merge the gradiente. I would interested in some kind of weighted add, like 0.2*gradient(left)+0.8*gradient(right). I suppose this is possible in code but I did't find it.
The thing is I would like to provide more importance to the training to learn from one side than the other

Tanks for your attention

**Comment by tboquet** [comment]
You can define a custom objective function for each output:

```python
def categorical_crossentropy_out(y_true, y_pred):
    '''Expects a binary class matrix instead of a vector of scalar classes.
    '''
    return 0.2 * K.categorical_crossentropy(y_pred, y_true)

def categorical_crossentropy_out2(y_true, y_pred):
    '''Expects a binary class matrix instead of a vector of scalar classes.
    '''
    return 0.8 * K.categorical_crossentropy(y_pred, y_true)

model.compile(rms,{'out':categorical_crossentropy_out,'out2':categorical_crossentropy_out2})
```

Is it what you are looking for?

**Comment by jllombart** [comment]
thank you, that it is!

i'm now trying other thing that i've found but seems harder, I will say you if it works
now i'm using weighted classes like here, each class is weighted with the same value, but different from classes from the other loss funtion
```
class_out_weight = dict([(i, 0.8) for i in range(out_classes)])
class_out2_weight = dict([(i, 0.1) for i in range(out2_classes)])

model.fit({'input':Xtr,
           'out':Ytr1,
           'out2':Ytr2},
            class_weight={'out':class_out_weight,
                         'out2':class_out2_weight},
            nb_epoch=10, batch_size=512, verbose=1)
```

I will try both and say here what I find
